src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -436,10 +434,74 @@ def _unpack_build(self, base_build_dir, build_dir, build_url):
       storage.copy_file_from(build_url, build_local_archive)
     except Exception as e:
       logs.error(f'Unable to download build from {build_url}: {e}')
-      return False
+      raise
 
     try:
       with build_archive.open(build_local_archive) as build:
+        yield build
+    finally:
+      shell.remove_file(build_local_archive)
+
+  def _open_build_archive(self, unpack_everything: Optional[bool],
+                          base_build_dir: str, build_dir: str, build_url: str,
+                          http_build_url: Optional[str]):
+    """Gets a handle on a build archive for the current build. Depending on the
+    provided parameters, this function might download the build archive into
+    the build directory or directly use remote HTTP archive.
+
+    Args:
+        base_build_dir: the base build directory.
+        build_dir: the current build directory.
+        build_url: the build URL.
+        http_build_url: the HTTP build URL.
+
+    Raises:
+        if an error occurred while accessing the file over HTTP or while
+        downloading the file on disk.
+
+    Returns:
+        the build archive.
+    """
+    # We only want to use remote unzipping if we're not unpacking everything and
+    # if the HTTP URL is compatible with remote unzipping.
+    can_unzip_over_http = not unpack_everything and http_build_url\
```

Comment:
I think we should probably gate this feature on an env var. We can add this env var to windows_libfuzzer_chrome_asan at first (which is already broken) and then expand it to more jobs or make it opt-out.